codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/AuthSchemeCodegenMetadata.java

```diff
@@ -87,35 +88,30 @@ public final class AuthSchemeCodegenMetadata {
 
     static final AuthSchemeCodegenMetadata BEARER = builder()
         .schemeId(BearerAuthScheme.SCHEME_ID)
+        .authSchemeClass(BearerAuthScheme.class)
         .build();
 
     static final AuthSchemeCodegenMetadata NO_AUTH = builder()
-        .schemeId(SelectedAuthScheme.SMITHY_NO_AUTH)
+        .schemeId(NoAuthAuthScheme.SCHEME_ID)
+        .authSchemeClass(NoAuthAuthScheme.class)
         .build();
 
     private final String schemeId;
     private final List<SignerPropertyValueProvider> properties;
+    private final Class<?> authSchemeClass;
 
     private AuthSchemeCodegenMetadata(Builder builder) {
         this.schemeId = Validate.paramNotNull(builder.schemeId, "schemeId");
         this.properties = Collections.unmodifiableList(Validate.paramNotNull(builder.properties, "properties"));
+        this.authSchemeClass = Validate.paramNotNull(builder.authSchemeClass, "authSchemeClass");
```

Comment:
nit: given this, it seems `.filter(Objects::nonNull)` here https://github.com/aws/aws-sdk-java-v2/blob/2a9dcf28b3158d87b70f03beff06ba95cf2f818a/codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/AuthSchemeSpecUtils.java#L195 can be removed